packages/legend-application-studio/src/components/editor/edit-panel/mapping-editor/PropertyMappingsEditor.tsx

```diff
@@ -230,120 +271,184 @@ export const PropertyMappingsEditor = observer(
+                          </button>
+                        )}
+                    </div>
+                  );
+                }
+                case SET_IMPLEMENTATION_TYPE.FLAT_DATA:
+                case SET_IMPLEMENTATION_TYPE.EMBEDDED_FLAT_DATA: {
+                  return (
+                    <div
+                      className="property-mapping-editor__entries__entry"
+                      key={propertyMappingState.uuid}
+                    >
+                      <FlatDataPropertyMappingEditor
+                        isReadOnly={isReadOnly}
+                        flatDataInstanceSetImplementationState={
+                          instanceSetImplementationState as FlatDataInstanceSetImplementationState
+                        }
+                        flatDataPropertyMappingState={
+                          propertyMappingState as FlatDataPropertyMappingState
+                        }
+                        setImplementationHasParserError={
+                          setImplementationHasParserError
+                        }
+                      />
+                      {propertyMappingStates.length > 1 &&
+                        propertyBasicType !== CLASS_PROPERTY_TYPE.CLASS && (
+                          <button
+                            key={propertyMappingState.uuid}
+                            className="property-mapping-editor__entries__entry__remove-btn"
+                            onClick={() =>
+                              removePropertyMapping(
+                                propertyMappingState.propertyMapping,
+                              )
+                            }
+                            tabIndex={-1}
+                            title="Remove"
+                            disabled={propertyMappingState.lambdaString === ''}
+                          >
+                            <TimesIcon />
+                          </button>
+                        )}
+                    </div>
+                  );
+                }
+                case SET_IMPLEMENTATION_TYPE.RELATIONAL: {
+                  return (
+                    <div
+                      className="property-mapping-editor__entries__entry"
+                      key={propertyMappingState.uuid}
+                    >
+                      <RelationalPropertyMappingEditor
+                        isReadOnly={isReadOnly}
+                        relationalInstanceSetImplementationState={
+                          instanceSetImplementationState as RootRelationalInstanceSetImplementationState
+                        }
+                        relationalPropertyMappingState={
+                          propertyMappingState as RelationalPropertyMappingState
+                        }
+                        setImplementationHasParserError={
+                          setImplementationHasParserError
+                        }
+                      />
+                      {propertyMappingStates.length > 1 &&
```

Comment:
Could propbably create a component to wrap thi and make this less duplicated I think. Something like:

```ts
<GenericPropertyMappingEditor><SomePropertyMappingEditor /></GenericPropertyMappingEditor>
```